### PR TITLE
RUBY 3.0.0 compatible, batch-loader updated

### PR DESCRIPTION
**Pull request by sanjsharma**
`def my_method(arg1, arg2: nil, arg3: nil)
   ...
 end`
 before ruby 3.0.0, `my_method('random', {})` used to work assuming nils for arg2 and arg3 ,
 but since ruby 3.0.0 it throws exceptions, expected 1 argument provided 2

### CODE REVIEW

File: lib/ams_lazy_relationships/core/relationship_wrapper_methods.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def define_lazy_association(type, name, options, block)
         end
       end
 
-      lazy_relationship(name, options.slice(*lazy_relationship_option_keys))
+      lazy_relationship(name, load_for: options[:load_for], loader: options[:loader])
```
Comment (stokarenko) [inline lib/ams_lazy_relationships/core/relationship_wrapper_methods.rb:47]:
```suggestion
      lazy_relationship(name, **options.slice(*lazy_relationship_option_keys))
```

**Comment by stokarenko** [comment]
Hey @sanjsharma, thank you for PR! )

Is there a chance to keep the support of `Ruby v2` in same time? The changes look not so devastating as I feel.. If so please keep rvm `2.5.3` in CI matrix )

Also there is some test fails against `Ruby v3`, looks like `thread_safe` appears to be out of bundle or something like that )

**Comment by Bajena** [comment]
Thanks for the PR @sanjsharma! 

I agree with @stokarenko:
1. We should try to keep both ruby 2 and ruby 3 tests. Check out https://docs.travis-ci.com/user/build-matrix/ - you can simply keep both `2.5.3` and `3.0.0` in the `rvm` section of the build config.
2. About tests failing because of missing `thread_safe`. I recently had a similar case when updating one of our apps to Rails 6. It turned out that in AMS `0.10.0.rc4` there's a `require 'thread_safe'` in AMS gem even though `thread_safe` isn't specified as a dependency in `gemspec`. I suppose that the AMS authors simply assumed that Rails will always include that dependency. Unfortunately in later versions Rails do not have `thread_safe` as a dependency because it became a part of `concurrent-ruby` 🙈 
3. There are other tests failing too - please have a look at the build (https://travis-ci.com/github/Bajena/ams_lazy_relationships/jobs/488804549)

**Comment by mike927** [comment]
Hello there,
when can we expect compatibility for ruby 3? Just updated our app to ruby3 but `ams_lazy_relationships` is the only one library that doesn't support it :( Getting mentioned error:
```
ArgumentError (wrong number of arguments (given 2, expected 1)):
```

**Comment by Bajena** [comment]
Hi @mike927 I'll try to get it done over the weekend:)

**Comment by Bajena** [comment]
@mike927 Ruby 3 compatibility released (v0.3.2). Thanks @sanjsharma for help here!

**Pull request closed without merging**